eZ/Publish/Core/FieldType/Keyword/KeywordStorage.php

```diff
@@ -26,7 +26,7 @@ class KeywordStorage extends GatewayBasedStorage
      */
     public function storeFieldData(VersionInfo $versionInfo, Field $field, array $context)
     {
-        if (empty($field->value->externalData)) {
+        if (empty($field->value->externalData) && !is_array($field->value->externalData)) {
```

Comment:
Well, the idea is to actually not assume that this value is wrong if it's empty. 
Double checked code paths and it seems that [with this change](https://github.com/alongosz/ezpublish-kernel/blob/c723fc2d888892030f38d2c033f501e0fc9800d8/eZ/Publish/Core/FieldType/Keyword/KeywordStorage/Gateway/LegacyStorage.php#L63-L67) this check is not required, so removed it completely at ed66ea7 - @andrerom maybe now it's better?

Once again, if `externalData` is empty that does not mean we can ignore storing it because we actually might want to override existing keyword list with empty list.